### ISSUE TITLE
explicit usage of `libp2p.dial()`/libp2p's autodial

### ISSUE THREAD

**Issue by danisharora099** [waku-org/js-waku#1117]
This is a change request

## Problem

We currently use `libp2p.dial()` to dial every one we discover. However, libp2p by default autodials to all nodes. (https://github.com/libp2p/js-libp2p/blob/master/doc/PEER_DISCOVERY.md)

The question now is:
- whether we want to manually handle dials (considering we can't find much information about the nodes by guarantee (eg protocols they support) and have to connect to them before we know whether its a useful node or not, it's good to auto connect)

## Proposed Solutions
Delete the usage of `libp2p.dial()` and let libp2p auto conect

## Notes

I'm not entirely sure if this is a good diea is why I haven't opened a PR for it yet, but just something that might open a discussion to help understand things better. (especially since we're considering a peer management overhaul)

**Comment by fryorcraken** [comment]
> However, libp2p by default autodials to all nodes. ([libp2p/js-libp2p@`master`/doc/PEER_DISCOVERY.md](https://github.com/libp2p/js-libp2p/blob/master/doc/PEER_DISCOVERY.md?rgh-link-date=2023-01-09T17%3A58%3A20Z))

Are you sure? https://github.com/libp2p/js-libp2p/blob/master/doc/migrations/v0.39-v0.40.md#autodial

**Comment by danisharora099** [comment]
> > However, libp2p by default autodials to all nodes. ([libp2p/js-libp2p@`master`/doc/PEER_DISCOVERY.md](https://github.com/libp2p/js-libp2p/blob/master/doc/PEER_DISCOVERY.md?rgh-link-date=2023-01-09T17%3A58%3A20Z))
> 
> Are you sure? I believe this was removed in 0.40.0 as documented in: https://github.com/libp2p/js-libp2p/blob/master/doc/migrations/v0.39-v0.40.md#autodial
> 
> If you remove
> 
> https://github.com/waku-org/js-waku/blob/0b083201c6f585ad1b3267c1571df5f1b81a1355/packages/core/src/lib/waku.ts#L140
> 
> Does it still auto-dial?

While it should autodial as can be seen by libp2p logs, the docs seem to be updated: 
```
  libp2p:connection-manager:auto-dialler options: {"enabled":true,"minConnections":50,"autoDialInterval":10000} +0ms
```
However, it I do not see dial attempts in the logs -- it is possible there is a timeout or delay with auto dial attempts.

I will close this issue for now based on our initial requirements of not needing to connect to all discovered peers and explicitly setting `autoDial` to false in the original issue.

Thanks!

**Comment by danisharora099** [comment]
> > However, libp2p by default autodials to all nodes. ([libp2p/js-libp2p@`master`/doc/PEER_DISCOVERY.md](https://github.com/libp2p/js-libp2p/blob/master/doc/PEER_DISCOVERY.md?rgh-link-date=2023-01-09T17%3A58%3A20Z))
> 
> Are you sure? I believe this was removed in 0.40.0 as documented in: https://github.com/libp2p/js-libp2p/blob/master/doc/migrations/v0.39-v0.40.md#autodial
> 
> If you remove
> 
> https://github.com/waku-org/js-waku/blob/0b083201c6f585ad1b3267c1571df5f1b81a1355/packages/core/src/lib/waku.ts#L140
> 
> Does it still auto dial?

Yes - I've been able to confirm the libp2p autodials discovered nodes by default as can be seen by the logs also:

```
  libp2p:connection-manager:auto-dialler options: {"enabled":true,"minConnections":50,"autoDialInterval":10000} +0ms
```

and autodial attempts:
```
  libp2p:connection-manager:auto-dialler connecting to a peerStore stored peer 16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm +10s
  libp2p:connection-manager dial to 16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm +10s
  libp2p:dialer check multiaddrs 16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm +0ms
  libp2p:dialer creating dial target for 16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm +0ms
  libp2p:peer-store:address-book:trace get wait for read lock +9s
  libp2p:peer-store:address-book:trace get got read lock +0ms
  libp2p:peer-store:address-book:trace get release read lock +0ms
  libp2p:dialer 1 tokens request, returning 1, 99 remaining +4ms
  libp2p:websockets dialing /dns4/node-01.ac-cn-hongkong-c.wakuv2.test.statusim.net/tcp/8000/wss/p2p/16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm +0ms
  libp2p:websockets dialing node-01.ac-cn-hongkong-c.wakuv2.test.statusim.net:8000 +1ms
```

```

**Comment by fryorcraken** [comment]
right. Thank you for this investigation. Which means the current dialign we do is useless. We should probably remove our manual dial.
When we tackle #914 we can disable auto dial and do things manually instead.

**Comment by danisharora099** [comment]
Sounds good.

I have addressed this issue here https://github.com/waku-org/js-waku/pull/1118

**Comment by danisharora099** [comment]
The inconsistency is also being confirmed here: https://github.com/libp2p/js-libp2p/pull/1397#issuecomment-1383934482

**Comment by danisharora099** [comment]
Update: It is concluded that libp2p by default autodials until `minConnections` and then gives control over dial attempts. However, for the scope of our project, we want complete control to be handled by Waku. Thus, disabling `autoDial`.

Ref: https://github.com/libp2p/js-libp2p/pull/1397#issuecomment-1383943063

**Comment by danisharora099** [comment]
This issue will be closed with the linked PR merge.

**Comment by fryorcraken** [comment]
which one is the linked PR?
